lua/astrocommunity/recipes/auto-session-restore/init.lua

```diff
@@ -0,0 +1,25 @@
+return {
+  "AstroNvim/astrocore",
+  ---@type AstroCoreOpts
+  opts = {
+    autocmds = {
+      -- disable alpha autostart
+      alpha_autostart = false,
+      restore_session = {
+        {
+          event = { "VimEnter", "StdinReadPost" },
+          desc = "Restore previous directory session if neovim opened with no arguments",
+          once = true, -- delete itself after executing once
+          nested = true, -- trigger other autocommands as buffers open
+          callback = function(args)
+            -- Only load the session if nvim was started with no args
+            if args.event == "VimEnter" and vim.fn.argc(-1) == 0 then
+              -- try to load a directory session using the current working directory
+              require("resession").load(vim.fn.getcwd(), { dir = "dirsession", silence_errors = true })
+            end
+          end,
```

Comment:
```suggestion
          nested = true, -- trigger other autocommands as buffers open
          callback = function(args)
            -- Only load the session if nvim was started with no args
            if args.event == "VimEnter" and vim.fn.argc(-1) == 0 then
              -- try to load a directory session using the current working directory
              require("resession").load(vim.fn.getcwd(), { dir = "dirsession", silence_errors = true })
            end
            return true -- delete itself after executing a single time
          end,
```

Autocommand callbacks can return `true` to indicate they should delete themselves.

Reply:
woah github diff algorithm so bad :joy:

Reply:
let's see if this works maybe, if not then we can always add in a `vim.api` call to just delete the `augroup`